src/base/lock_free.c

```diff
@@ -893,6 +893,20 @@ lf_freelist_retire (LF_TRAN_ENTRY * tran_entry, LF_FREELIST * freelist, void *en
 	}
     }
 
+  /* free entry if alloc_cnt is greater than max_alloc_cnt */
+  if (freelist->alloc_cnt > edesc->max_alloc_cnt)
+    {
+      edesc->f_free (entry);
```

Comment:
To return the freed memory to the OS without keeping it in the heap, `malloc_trim (0);` must be called.
However, because it is called once per LOCK, the performance of `lf_freelist_retire` decreases.
If the performance of `lf_freelist_retire` decreases, the performance of COMMIT also decreases.

Reply:
This issue is intended to defend against cases where OOM has occurred in the past. To prevent OOM, I think it's enough for the OS to recognize the memory as reusable.